hazelcast-client/src/test/java/com/hazelcast/client/cardinality/ClientCardinalityEstimatorTest.java

```diff
@@ -0,0 +1,220 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.cardinality;
+
+import com.hazelcast.cardinality.CardinalityEstimator;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.config.SerializerConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.nio.Bits;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.StreamSerializer;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class ClientCardinalityEstimatorTest
+        extends HazelcastTestSupport {
+
+    private final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
+    private HazelcastInstance client;
+    private CardinalityEstimator estimator;
+
+    @Before
+    public void setup() {
+        hazelcastFactory.newHazelcastInstance();
+        client = hazelcastFactory.newHazelcastClient();
+        estimator = client.getCardinalityEstimator(randomString());
+    }
+
+    @After
+    public void tearDown() {
+        hazelcastFactory.terminateAll();
+    }
+
+    private HazelcastInstance createSerializationConfiguredClient() {
+        final SerializerConfig serializerConfig = new SerializerConfig();
+        serializerConfig.setImplementation(new CustomObjectSerializer());
+        serializerConfig.setTypeClass(CustomObject.class);
+
+        ClientConfig config = new ClientConfig();
+        config.getSerializationConfig().addSerializerConfig(serializerConfig);
+
+        return hazelcastFactory.newHazelcastClient(config);
+    }
+
+    @Test
+    public void test() throws Exception {
+        assertEquals(0, estimator.estimate());
+        assertEquals(true, estimator.aggregate(1L));
+        assertEquals(true, estimator.aggregate(1L));
+        assertEquals(1, estimator.estimate());
+
+        for (long l : new long[] { 2L, 3L, 4L }) {
+            assertEquals(true, estimator.aggregate(l));
+        }
+
+        assertEquals(4, estimator.estimate());
+        assertEquals(true, estimator.aggregate("Test"));
+        assertEquals(5, estimator.estimate());
+    }
+
+    @Test
+    public void testAsync() throws Exception {
+        ICompletableFuture<Long> f1 = estimator.estimateAsync();
```

Comment:
I'm not that crazy about these do/verify/do/verify tests. 

Normally best to split them up to individual test methods. If they need to have an assertion. Otherwise one quickly gets dirty test since it isn't obvious if a particular case is tested, so it is being repeated all over.